src/Core/Checkout/Promotion/DataAbstractionLayer/PromotionRedemptionUpdater.php

```diff
@@ -104,54 +114,20 @@ public function update(array $ids, Context $context): void
         }
     }
 
-    public function orderPlaced(CheckoutOrderPlacedEvent $event): void
+    public function orderUpdated(EntityWrittenEvent $event): void
     {
-        $lineItems = $event->getOrder()->getLineItems();
-        $customer = $event->getOrder()->getOrderCustomer();
-
-        if (!$lineItems || !$customer) {
+        if ($event->getEntityName() !== OrderDefinition::ENTITY_NAME) {
             return;
         }
 
```

Comment:
We shouldn't do this only on live versions, right?
```suggestion
if ($event->getContext()->getVersionId() !== Defaults::LIVE_VERSION) {
    continue;
}
```

Reply:
your English is really wrong here and contradictory to your code suggestion 😉